components/omega/src/base/Halo.cpp

```diff
@@ -506,13 +506,12 @@ int Halo::packBuffer(const ArrayHost1DR8 Array) {
    }
 
    return 0;
-} // end packBuffer ArrayHost1DR8
+} // end packBuffer HostArray1DR8
 
-int Halo::packBuffer(const ArrayHost2DI4 Array) {
+int Halo::packBuffer(const HostArray2DI4 Array) {
 
-   ExchList *MyList  = &MyNeighbor->SendLists[MyElem];
-   yakl::Dims MyDims = Array.get_dimensions();
-   int NJ            = MyDims[1];
+   ExchList *MyList = &MyNeighbor->SendLists[MyElem];
+   int NJ           = Array.extent(0);
```

Comment:
Typo here is causing the segfaults, should be `Array.extent(1);`

Reply:
@brian-oneill Thanks for finding this bug that I introduced. After fixing this,  I successfully ran Halo test on Frontier with HIP. Once I complete Halo test on other machines, I will update this PR.